inferred-spans/src/main/java/io/opentelemetry/contrib/inferredspans/InferredSpansSpanProcessorProvider.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.inferredspans;
+
+import com.google.auto.service.AutoService;
+import io.opentelemetry.api.incubator.config.DeclarativeConfigProperties;
+import io.opentelemetry.instrumentation.config.bridge.DeclarativeConfigPropertiesBridgeBuilder;
+import io.opentelemetry.sdk.autoconfigure.spi.internal.ComponentProvider;
+import io.opentelemetry.sdk.trace.SpanProcessor;
+
+@SuppressWarnings("rawtypes")
+@AutoService(ComponentProvider.class)
+public class InferredSpansSpanProcessorProvider implements ComponentProvider<SpanProcessor> {
+
+  private static final String PREFIX = "otel.inferred.spans.";
+
+  @Override
+  public String getName() {
+    return "experimental_inferred_spans";
```

Comment:
so clearly from the discussion in the README, we'll want to support more than one name. How will that be achieved (specifically, we'll want both `experimental_inferred_spans` and `inferred_spans` to be supported at some point in the future)

Reply:
So a duplicated class (or would a subclass work?), with either one or the other installed depending on the declaration. I'm good with that, will that work @zeitlinger

Reply:
> @zeitlinger for naming, what about `inferred_spans/development` to match the `*/development` convention for declarative configuration property names?

@jack-berg pointed out that we should use the `experimental_*` convention

Reply:
@jack-berg please help me out here

Reply:
> so clearly from the discussion in the README, we'll want to support more than one name.

let's make this a separate discussion - I've added it to the Java SIG agenda

Reply:
> The name isn't my focus here, it's being able to support both names indefinitely when this becomes stable

this is why I added it to the meeting agenda